src/shared/cpu-set-util.c

```diff
@@ -62,24 +63,74 @@ cpu_set_t* cpu_set_malloc(unsigned *ncpus) {
         }
 }
 
-int parse_cpu_set_internal(
+static int cpu_set_realloc(CPUSet *cpu_set, unsigned ncpus) {
+        size_t need;
+
+        assert(cpu_set);
+
+        need = CPU_ALLOC_SIZE(ncpus);
```

Comment:
hmm, maybe use this instead:

```c
need = CPU_ALLOC_SIZE(MAX(ncpus, get_nprocs_conf()));
```

i.e. it might be a good idea to start out with the number of cpus in the system here

Reply:
If we are on a large machine, we might do something like `CPUAffinity=0-3`, and then we are fine with the minimum allocation. I'd just leave things simple here.